File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int recur_climbStairs(int n, vector<int> &memo) {
        if (memo[n] != 0) {
            return memo[n];
        }
        memo[n] = recur_climbStairs(n-1, memo) + recur_climbStairs(n-2, memo);
        return memo[n];
    }
    
    int climbStairs(int n) {
        if (n == 1) {
            return 1;
        }
        else if (n == 2) {
            return 2;
        }
        vector<int> memo(n+1);
        memo[0] = 1;
        memo[1] = 1;
        memo[2] = 2;
        return recur_climbStairs(n, memo);
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    @lru_cache(maxsize=None)
    def climbStairs(self, n: int) -> int:
        # base case
        if (n == 1):
            return 1
        elif (n == 0):
            return 1
    
        # recursive case
        return self.climbStairs(n-1) + self.climbStairs(n-2)
        